### ISSUE TITLE
Feature : Revamping/Enhancing the Groups Page.

### ISSUE THREAD

**Issue by Asrani-Aman** [CircuitVerse/CircuitVerse#4771]
### Is your feature request related to a problem? Please describe

The Groups page requires revamping due to the following reasons:

1. Currently, users need to click on the "view button" to access a Group, which could be streamlined by allowing users to navigate directly to a Group by clicking on its card.

2. add description section for a group

3. Lack of search functionality makes it challenging for users, especially educators with multiple groups across various sections or institutes, to locate specific groups efficiently.
4. The design of the group cards needs to be updated to align with modern aesthetics.

### Describe the solution you'd like

proposing  a Figma Design for the revamp of the Groups Page, encompassing the following features:

1. Integration of search functionality to facilitate easy navigation and discovery of groups.
2. Modernisation of icons and card design to enhance visual appeal and user experience.
3. Addition of a description section for each group to provide context and relevant information.
4. Implementation of a hovering effect on group cards for better interactivity. .
5. Enable direct navigation to a specific group by clicking on its card.

here is the[ Figma Design](https://www.figma.com/file/K2VtkUdgXvqMIUn3XRCXAd/Description-Box?type=design&node-id=0%3A1&mode=design&t=64lfXxK4X0E2n8ma-1)

### Are you working on this?

Yes

**Comment by Asrani-Aman** [comment]
@tanmoysrt bhaiya please review this issue , and please suggest any improvements or modifications

**Comment by tanmoysrt** [comment]
Hi @Asrani-Aman you can work on this.

@vedant-jain03 have you any feedback for the proposed design ?

**Comment by Asrani-Aman** [comment]
hello @Kiran-pande-30 , i have also proposed the group page in the UX channel earlier along with groups page ,

you can check it in the same [Figma Design link](https://www.figma.com/file/K2VtkUdgXvqMIUn3XRCXAd/Description-Box?type=design&node-id=0%3A1&mode=design&t=ysPkeNuYdKJV7DJU-1) while you scroll you will find the same

**Comment by Kiran-pande-30** [comment]
@Asrani-Aman Okay, but can now also review my design, as it will help me in  my further UX Designs

**Comment by Asrani-Aman** [comment]
yeah sure @Kiran-pande-30 , but i am also not much professional in Figma, but as my point of view overall Figma design looks decent , generally i do make Figma designs by taking inspirations from different websites (one of the example is [Dribble](https://dribbble.com))

**Comment by Asrani-Aman** [comment]
for the  groups page , i first see the similar platforms (like google classroom ) and then see various cards design on internet and Figma templates. this will help a lot in making Figma designs  😇

**Comment by Kiran-pande-30** [comment]
Thanks for the insights @Asrani-Aman
